### ISSUE TITLE
TypeError: __int__ returned non-int (type NoneType)

### ISSUE THREAD

**Issue by frogg** [calclavia/DeepJ#61]
After installing all requirements (`python-midi` and `requirements.txt`) I get this error message when executing `generate.py`:

```
Using TensorFlow backend.
Traceback (most recent call last):
  File "generate.py", line 153, in <module>
    main()
  File "generate.py", line 142, in main
    models = build_or_load()
  File "/Users/frederikriedel/Developer/DeepJ/util.py", line 15, in build_or_load
    models = build_models()
  File "/Users/frederikriedel/Developer/DeepJ/model.py", line 149, in build_models
    notes_out = naxis(time_out, chosen, style)
  File "/Users/frederikriedel/Developer/DeepJ/model.py", line 111, in f
    dense_layer_cache[l] = Dense(int(x.get_shape()[3]))
TypeError: __int__ returned non-int (type NoneType)
```

Do you maybe know what I'm missing here?

**Comment by calclavia** [comment]
Hmm, it could be a Tensorflow version mismatch? Maybe you're using a newer version than the one DeepJ was developed in?

**Comment by frogg** [comment]
That was a great hint. It wasn't the tensorflow version itself that caused the issue, but keras. 
I installed an older version of keras now and it seems to work!

**Comment by calclavia** [comment]
@frogg Great to know. If you want , feel free to submit a PR to update our code to the latest Keras so others won't run into the same issue. Seems like it would be a one-line change.

**Comment by dannywu19910524** [comment]
@frogg thanks a lot, your re help me

**Comment by swarna-a-26** [comment]
@frogg @calclavia I am using keras 2.1.5 and I got the same error. Which version of Keras should I use? Please let me know.

**Comment by swarna-a-26** [comment]
@frogg  Could you please let me know the tensorflow and keras version.

**Comment by frogg** [comment]
@swarna-a-26 I think I was using keras 1.2.1.
Let me know if that works. Maybe @dannywu19910524 remembers it better than me.

**Comment by swarna-a-26** [comment]
@frogg @dannywu19910524 I used tensorflow1.6.0 and keras 2.0.0.That seems to remove the above error.Thanks so much. 
I am getting the below error when I try to train the data.
volume_dense (Dense)             multiple              129                                          
____________________________________________________________________________________________________
concatenate_3 (Concatenate)      (None, 128, 48, 3)    0                                            
====================================================================================================
Total params: 1,269,476.0
Trainable params: 1,269,476.0
Non-trainable params: 0.0
____________________________________________________________________________________________________
Unable to load model from file.
Loading data
Training
Traceback (most recent call last):
  File "train.py", line 32, in <module>
    main()
  File "train.py", line 16, in main
    train(models)
  File "train.py", line 29, in train
    models[0].fit(train_data, train_labels, epochs=1000, callbacks=cbs, batch_size=BATCH_SIZE)
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 1405, in fit
    batch_size=batch_size)
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 1295, in _standardize_user_data
    exception_prefix='model input')
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 121, in _standardize_input_data
    str(array.shape))
ValueError: Error when checking model input: expected input_1 to have 4 dimensions, but got array with shape (0, 1)

Any idea about this error? I created a data folder and have three mid files there. Any thoughts to overcome this issue? Is this one also related to keras and tensorflow version? any thoughts will be helpful.I am using MAC 10.13.1

**Comment by dannywu19910524** [comment]
@swarna-a-26 i didn't get this error i'll check my keras and tensorflow version later for u

**Comment by dannywu19910524** [comment]
@swarna-a-26 tensorflow-gpu 1.2 keras 2.0

**Comment by slave2sync** [comment]
can you please recommend a way out of the error @swarna-a-26 mentioned?
I've used python3, tensorflow 1.2.0 (no GPU), keras 2.0 and also got 
ValueError: Error when checking model input: expected input_1 to have 4 dimensions, but got array with shape (0, 1)

please let us know if you can

**Comment by JucyCherry** [comment]
@slave2sync @swarna-a-26 Have you solve the problem? And can you share the way out of the error:
ValueError: Error when checking model input: expected input_1 to have 4 dimensions, but got array with shape (0, 1)

**Comment by soundreactor** [comment]
any news?

**Comment by ashokrs** [comment]
@calclavia @slave2sync @swarna-a-26
I am on Ubtuntu Linux with  TF 1.12 with GPU (without GPU) and Keras 2.0.
I am getting the same ValueError:  Error when checking model input: expected input_1 to have 4 dimensions, but got array with shape (0, 1)

Any help would be much appreciated?

**Comment by amitjslearn** [comment]
> I am getting the same ValueError: Error when checking model input: expected input_1 to have 4 dimensions, but got array with shape (0, 1)

Hey, I am also getting the same error. If you fixed it please, let me know.
Thank you.

**Comment by KDcx35** [comment]
> > I am getting the same ValueError: Error when checking model input: expected input_1 to have 4 dimensions, but got array with shape (0, 1)
> 
> Hey, I am also getting the same error. If you fixed it please, let me know.
> Thank you.

Hey,I had the same error before,than i solved it thanks to this issue #72
You need to construct the training data fold like the styles and genres in 'constant.py', than the error would disappear(at least for me).
Hope this would help!
